### PR TITLE
Wi 48 photo quick view cindy

### PR DESCRIPTION
**Pull request by Cindy088**

### CODE REVIEW

File: pages/admin-manager.tsx [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 import Main from "../src/components/AdminManage/Main";
 import Copyright from "../src/components/Copyright";
 
-// Define a component that renders the form
+// Define a component that renders the page
 export default function AdminManager() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    const accessToken = window.localStorage.getItem("accessToken");
```
Comment (Chanran) [inline pages/admin-manager.tsx:12]: Move setItem and getItem code into the storage.ts file.

File: pages/photo-quick-view/[filename].tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+  const router = useRouter();
+  const { filename } = router.query;
+
+  // fetch post user avatar,username,post image
+  useEffect(() => {
+    if (!router.isReady) return;
+    const fetchData = async () => {
+      try {
+        const response = await axiosRequest(
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx:38]: Move the request code in the "photo.ts" file in the api folder

File: pages/photo-quick-view/[filename].tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+          `/api/quickView?filename=${filename}`,
+          "GET"
+        );
+        if (!response) {
+          return;
+        }
+        setUserName(response.data.user_name);
+        setUserAvatar(response.data.avatar_url);
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx:43]: text indent of code is not correct

File: pages/photo-quick-view/[filename].tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        setPostPhoto(response.data.photo_url);
+        setUserCollects(response.data.collects);
+        setUserLikes(response.data.likes);
+      } catch (error: any) {
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx:50]: do not use any

File: pages/photo-quick-view/[filename].tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  //click close button to redirect back to home page
+  const onClickCloseButton = () => {
+    router.push("/");
+  };
+
+  //Toggle collect button and add/delete collect number
+  const addToCollection = async () => {
+    //click collect button to update number
+    try {
+      const response = await axiosRequest(`/api/collect/${filename}`, "POST", {
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx:61]: Move the request code in the "photo.ts" file in the api folder

File: pages/photo-quick-view/[filename].tsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    } catch (error: any) {
+      router.push("/login");
+    }
+  };
+
+  //Toggle like button and add/delete like number
+  const addToLiked = async () => {
+    //click like button to update number
+    try {
+      const response = await axiosRequest(`/api/like/${filename}`, "POST", {
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx:75]: Move the request code in the "photo.ts" file in the api folder

File: pages/photo-quick-view/[filename].tsx [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+      const data = response.data;
+      setUserLikes(data);
+      setLiked((liked) => !liked);
+    } catch (error: any) {
+      router.push("/login");
+    }
+  };
+
+  //Redirect to download page
+  const downLoadImages = () => {};
```
Comment (Chanran) [inline pages/photo-quick-view/[filename].tsx]: Empty function？

File: src/components/AdminManage/Header.tsx [hunk 8]
```diff
@@ -6,12 +7,19 @@ import { useRouter } from "next/router";
 //Render header part of admin manager page
 const Header = () => {
   const router = useRouter();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
```
Comment (Chanran) [inline src/components/AdminManage/Header.tsx:10]: The loggin logic can be reused. You write same code in the Header.tsx and admin-manager.tsx

File: src/components/AdminManage/Header.tsx [hunk 9]
```diff
@@ -6,12 +7,19 @@ import { useRouter } from "next/router";
...
+
+  useEffect(() => {
+    const accessToken = window.localStorage.getItem("accessToken");
+    accessToken !== null ? setIsLoggedIn(true) : setIsLoggedIn(false);
+  }, []);
 
   //remover access token to redirect to admin sign in page
   const logOutHandler = () => {
-    window.localStorage.removeItem("access Token");
+    window.localStorage.removeItem("accessToken");
```
Comment (Chanran) [inline src/components/AdminManage/Header.tsx:19]: Move to code to "storage.ts"

**Review by JeremyyyLiu — APPROVED** [review]
Approve to merge

**Pull request merged**